tensorflow/contrib/go/graph_test.go

```diff
@@ -0,0 +1,252 @@
+package tensorflow_test
+
+import (
+	"testing"
+
+	tf "github.com/tensorflow/tensorflow/tensorflow/contrib/go"
+)
+
+func TestGraphGeneration(t *testing.T) {
+	graph := tf.NewGraph()
+	input1 := graph.Placeholder("input1", tf.DtInt32, []int64{3}, []string{})
```

Comment:
I wonder if it would be worth changing the datatype convention to DTInt32 / etc. instead of Dt -- the type is named DataType, so it might be a more natural contraction.  Up to you - I don't feel really strongly about this,.